eth/gasprice/gasprice.go

```diff
@@ -201,18 +203,66 @@ func (gpo *Oracle) getBlockPrices(ctx context.Context, signer types.Signer, bloc
 	for _, tx := range txs {
 		sender, err := types.Sender(signer, tx)
 		if err == nil && sender != block.Coinbase() {
-			prices = append(prices, tx.GasPrice())
+			// Remove transactions with a gasPrice of 0
+			if tx.GasPrice() != big.NewInt(0) {
+				prices = append(prices, tx.GasPrice())
+			}
 			if len(prices) >= limit {
 				break
 			}
 		}
 	}
+	if len(prices) > 0 {
+		prices = removeOutliers(prices)
+	}
 	select {
 	case result <- getBlockPricesResult{prices, nil}:
 	case <-quit:
 	}
 }
 
+// removeOutliers calculates the interquartile range of gas prices that are
+// significant outliers with the goal of removing edge cases where MEV skews
+// the gas prices
+func removeOutliers(prices []*big.Int) []*big.Int {
+	var (
+		mean       *big.Int
+		sd         *big.Int // Standard deviation
+		variance   = big.NewInt(0)
+		sum        = big.NewInt(0)
+		length     = big.NewInt(int64(len(prices)))
+		deviations = big.NewInt(3) // The max acceptable std, anything outsid will be removed from the set
+	)
+	for _, price := range prices {
+		// Calculate sum
+		sum.Add(sum, price)
+	}
+	mean = big.NewInt(0).Div(sum, length)
+	// Calculate variance (sum(x - mean)^2 )/ length
```

Comment:
`// Calculate variance (sum((x - mean)^2)/ length)` ?

Reply:
Woops this is old.